samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                        Marshal.ThrowExceptionForHR(hr);
+                    }
+
+                    // Wait for the command list to execute; we are reusing the same command
+                    // list in our main loop but for now, we just want to wait for setup to
+                    // complete before continuing.
+                    WaitForPreviousFrame();
+                }
+            }
+            finally
+            {
+                if (signature != null)
+                {
+                    signature->Release();
+                }
+
+                if (error != null)
+                {
+                    error->Release();
+                }
+
+                if (vertexShader != null)
+                {
+                    vertexShader->Release();
+                }
+
+                if (pixelShader != null)
+                {
+                    pixelShader->Release();
+                }
+            }
+        }
+
+        /// <summary> Fill the command list with all the render commands and dependent state. </summary>
+        private void PopulateCommandList()
+        {
+            // Command list allocators can only be reset when the associated
+            // command lists have finished execution on the GPU; apps should use
+            // fences to determine GPU execution progress.
+            ThrowIfFailed(nameof(ID3D12CommandAllocator.Reset), _commandAllocator->Reset());
+
+            // However, when ExecuteCommandList() is called on a particular command
+            // list, that command list can then be reset at any time and must be before
+            // re-recording.
+            ThrowIfFailed(nameof(ID3D12GraphicsCommandList.Reset), _commandList->Reset(_commandAllocator, _pipelineState));
+
+            // Set necessary state.
+            _commandList->SetGraphicsRootSignature(_rootSignature);
+
+            var pHeap = _cbvHeap;
+            _commandList->SetDescriptorHeaps(1, &pHeap);
+            var handle = _cbvHeap->GetGPUDescriptorHandleForHeapStart();
+            _commandList->SetGraphicsRootDescriptorTable(0, handle);
+
+            fixed (D3D12_VIEWPORT* viewport = &_viewport)
+            {
+                _commandList->RSSetViewports(1, viewport);
+            }
+
+            fixed (RECT* scissorRect = &_scissorRect)
+            {
+                _commandList->RSSetScissorRects(1, scissorRect);
+            }
+
+            // Indicate that the back buffer will be used as a render target.
+            var barrier = D3D12_RESOURCE_BARRIER.InitTransition(_renderTargets[unchecked((int)_frameIndex)], D3D12_RESOURCE_STATE_PRESENT, D3D12_RESOURCE_STATE_RENDER_TARGET);
+            _commandList->ResourceBarrier(1, &barrier);
+
+            var rtvHandle = _rtvHeap->GetCPUDescriptorHandleForHeapStart();
+            rtvHandle.ptr = (UIntPtr)((byte*)rtvHandle.ptr + (_frameIndex * _rtvDescriptorSize));
+            _commandList->OMSetRenderTargets(1, &rtvHandle, FALSE, null);
```

Comment:
This can likely be simplified to:
```suggestion
            var rtvHandle = new D3D12_CPU_DESCRIPTOPR_HANDLE(_rtvHeap->GetCPUDescriptorHandleForHeapStart(), _frameINdex, _rtvDescriptorSize);
            _commandList->OMSetRenderTargets(1, &rtvHandle, FALSE, null);
```

The HelloTriangle sample was ported before the above helper was and this was likely missed in the cleanup.